include/nlohmann/json.hpp

```diff
@@ -1133,51 +1133,53 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
 
         void destroy(value_t t) noexcept
         {
-            // flatten the current json_value to a heap-allocated stack
-            std::vector<basic_json> stack;
-
-            // move the top-level items to stack
-            if (t == value_t::array)
-            {
-                stack.reserve(array->size());
-                std::move(array->begin(), array->end(), std::back_inserter(stack));
-            }
-            else if (t == value_t::object)
-            {
-                stack.reserve(object->size());
-                for (auto&& it : *object)
-                {
-                    stack.push_back(std::move(it.second));
-                }
-            }
-
-            while (!stack.empty())
+            if (t == value_t::array || t == value_t::object)
```

Comment:
Since this function is now used in functions that are not `noexcept`, and it uses a `std::vector`, I don't think it should be marked `noexcept` any more.  It made sense when it was only used from the destructor.

Reply:
You're right - even before my change the function was not `noexcept` because of the vector. I'll remove the annotation.

Reply:
It was okay before, since it was only called from a destructor, and throwing in a destructor-called function is no different than throwing in the destructor if there is no catch for it.

Reply:
You're right. So what do you think is better in this situation:

1. Remove the `noexcept`.
2. Surround the function with a `try`/`catch(...)` block to enforce the ´noexcept`?

Reply:
Remove the noexcept.  The library shouldn't try to swallow the exception.